src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -245,48 +245,186 @@ Or, ``source`` the script and run the tests manually::
 How to add a new controller?
```

Comment:
Instead of a question, I suggest to rephrase this header to "Adding a New Controller"